typescript/core/src/v0/Mcap0IndexedReader.test.ts

```diff
@@ -245,14 +245,35 @@ describe("Mcap0IndexedReader", () => {
       data: new Uint8Array(),
     };
     it.each([
-      { startTime: undefined, endTime: undefined, expected: [message1, message2, message3] },
-      { startTime: 11n, endTime: 11n, expected: [message2] },
-      { startTime: 11n, endTime: undefined, expected: [message2, message3] },
-      { startTime: undefined, endTime: 11n, expected: [message1, message2] },
-      { startTime: 10n, endTime: 12n, expected: [message1, message2, message3] },
+      {
+        startTime: undefined,
+        endTime: undefined,
+        expected: [message1, message2, message3],
+        reverse: false,
+      },
+      { startTime: 11n, endTime: 11n, expected: [message2], reverse: false },
+
+      { startTime: 11n, endTime: undefined, expected: [message2, message3], reverse: false },
+      { startTime: undefined, endTime: 11n, expected: [message1, message2], reverse: false },
+      { startTime: 10n, endTime: 12n, expected: [message1, message2, message3], reverse: false },
+      {
+        startTime: undefined,
+        endTime: undefined,
+        expected: [message3, message2, message1],
+        reverse: true,
+      },
+      { startTime: 11n, endTime: 11n, expected: [message2], reverse: true },
+      { startTime: 11n, endTime: undefined, expected: [message3, message2], reverse: true },
+      {
+        startTime: undefined,
+        endTime: 11n,
+        expected: [message2, message1],
+        reverse: true,
+      },
+      { startTime: 10n, endTime: 12n, expected: [message3, message2, message1], reverse: true },
     ])(
-      "fetches chunk data and reads requested messages between $startTime and $endTime",
-      async ({ startTime, endTime, expected }) => {
+      "fetches chunk data and reads requested messages between $startTime and $endTime, reverse: $reverse",
+      async ({ startTime, endTime, expected, reverse }) => {
```

Comment:
I would remove reverse from the test params, and just run two `expect`s within each test. The `expected` result when reverse=true should always be reversed from the original expected result.